metasrv/src/network.rs

```diff
@@ -22,30 +23,61 @@ use async_raft::raft::InstallSnapshotResponse;
 use async_raft::raft::VoteRequest;
 use async_raft::raft::VoteResponse;
 use async_raft::RaftNetwork;
+use common_containers::ItemManager;
+use common_containers::Pool;
 use common_meta_types::LogEntry;
 use common_meta_types::NodeId;
 use common_tracing::tracing;
+use tonic::client::GrpcService;
 use tonic::transport::channel::Channel;
 
 use crate::proto::meta_service_client::MetaServiceClient;
 use crate::store::MetaRaftStore;
 
+struct ChannelManager {}
+
+#[async_trait]
+impl ItemManager for ChannelManager {
+    type Key = String;
+    type Item = Channel;
+    type Error = tonic::transport::Error;
+
+    async fn build(&self, addr: &Self::Key) -> Result<Channel, tonic::transport::Error> {
+        tonic::transport::Endpoint::new(addr.clone())?
+            .connect()
+            .await
+    }
+
+    async fn check(&self, mut ch: Channel) -> Result<Channel, tonic::transport::Error> {
+        futures::future::poll_fn(|cx| (&mut ch).poll_ready(cx)).await?;
+        Ok(ch)
+    }
+}
+
 pub struct Network {
     sto: Arc<MetaRaftStore>,
+
+    conn_pool: Pool<ChannelManager>,
 }
 
 impl Network {
     pub fn new(sto: Arc<MetaRaftStore>) -> Network {
-        Network { sto }
+        let mgr = ChannelManager {};
+        Network {
+            sto,
+            conn_pool: Pool::new(mgr, Duration::from_millis(50)),
```

Comment:
remove the local variable and just pass `ChannelManager{}` to new here?

Reply:
🤔 I'd hate to see more than one things done in one line...